File: SDKs/i-cue.solutions/v1/python/src/sdk/models/operations/post_forecast_optimal_parameter.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class PostForecastOptimalParameterHeaders:
    token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'Token', 'style': 'simple', 'explode': False }})
    

@dataclass
class PostForecastOptimalParameterRequests:
    planning_level_request: Optional[shared.PlanningLevelRequest] = field(default=None, metadata={'request': { 'media_type': 'application/*+json' }})
    planning_level_request1: Optional[shared.PlanningLevelRequest] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    planning_level_request2: Optional[shared.PlanningLevelRequest] = field(default=None, metadata={'request': { 'media_type': 'text/json' }})
    

@dataclass
class PostForecastOptimalParameterRequest:
    headers: PostForecastOptimalParameterHeaders = field()
    request: Optional[PostForecastOptimalParameterRequests] = field(default=None)
    

@dataclass
class PostForecastOptimalParameterResponse:
    content_type: str = field()
    status_code: int = field()
    body: Optional[bytes] = field(default=None)
    optimal_parameter_response: Optional[shared.OptimalParameterResponse] = field(default=None)
    

File: SDKs/i-cue.solutions/v1/typescript/src/sdk/models/operations/postforecastoptimalparameter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostForecastOptimalParameterHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Token" })
  token?: string;
}


export class PostForecastOptimalParameterRequests extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/*+json" })
  planningLevelRequest?: shared.PlanningLevelRequest;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  planningLevelRequest1?: shared.PlanningLevelRequest;

  @SpeakeasyMetadata({ data: "request, media_type=text/json" })
  planningLevelRequest2?: shared.PlanningLevelRequest;
}


export class PostForecastOptimalParameterRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: PostForecastOptimalParameterHeaders;

  @SpeakeasyMetadata()
  request?: PostForecastOptimalParameterRequests;
}


export class PostForecastOptimalParameterResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  optimalParameterResponse?: shared.OptimalParameterResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
